web/src/scheduled_messages.js

```diff
@@ -15,34 +15,71 @@ import * as narrow from "./narrow";
 import * as people from "./people";
 import * as popover_menus from "./popover_menus";
 import * as stream_data from "./stream_data";
+import * as timerender from "./timerender";
 
 export let scheduled_messages_data = [];
 
+const st = compute_later_send_times();
+
+function compute_later_send_times(now = new Date()) {
+    const send_times = {};
+    const today = new Date(now);
+    const tomorrow = new Date(new Date(now).setDate(now.getDate() + 1));
+    // Find the next Monday by subtracting the current day (0-6) from 8
+    const monday = new Date(new Date(now).setDate(now.getDate() + 8 - now.getDay()));
+    // today at 9am
+    send_times.today_nine_am = today.setHours(9, 0, 0, 0);
+    // today at 4pm
+    send_times.today_four_pm = today.setHours(16, 0, 0, 0);
```

Comment:
Hmm, does this work? I would think https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setHours would change the existing date object, not generate a new one, so these would all have the same value.

Reply:
It works as a side-effect: calling `setHours()` returns the value as a Unix timestamp (in addition to of course changing the value of `today` in place, as you note). The code isn't actually doing anything with the `today` Date object itself except to pull off those stamps and store them on `send_times`, in other words.

Reply:
OK. This is exactly the type of subtle detail that looks like it might be a bug and thus is worth adding a comment for; added one so I can merge this without an extra review round trip.

Reply:
Noted for the future. Thanks!